### ISSUE TITLE
PTX subcloud loading steals focus

### ISSUE THREAD

**Issue by ibtaylor** [cloudcompare/trunk#275]
In many window managers, new windows get focus. Every time a `ptx` subcloud loads, a new progress window is created and this steals focus away from my current window. I'm loading very large ptx files and so this effectly prevents me from working on other things while this loads.

Can this window be created differently so this behavior does not continue?

I essentially had to `setAutoClose(false)` and update the FileIOFilter api to receive the `ccProgressDialog`.

I made a patch that works for me, but it seemed like an inappropriate fix as it couples gui components with file loaders. Maybe the io filters should have various callbacks instead?

Thanks!

**Comment by cloudcompare** [comment]
In fact in the particular case of File I/O filters the progress dialog is generally created by the filter itself. Therefore we can easily create the PTX progress dialog only once and 'reset' it for each scan.

And I added setAutoClose(false) to the PTX progress dialog.

Another issue was that previously the filters had no connection to the application window. I modified the code so that now the created dialogs are children of the parent window (this way the dialogs should always appear on top of CC's main window and not over the window which has the focus).

I'll let you test this on your side.
